app/helpers/react_on_rails_helper.rb

```diff
@@ -86,14 +86,17 @@ def debug_js(react_component, data_variable, dom_id, trace)
     end
   end
 
+  def render_js_react_element(react_component, props)
+    ReactOnRails::ReactRenderer.new.render_js_react_element(react_component, props).html_safe
+  end
+
   def define_render_if_dom_node_present(react_component, data_variable, dom_id, trace)
     <<-JS.strip_heredoc
       var renderIfDomNodePresent = function() {
         var domNode = document.getElementById('#{dom_id}');
         if (domNode) {
           #{debug_js(react_component, data_variable, dom_id, trace)}
-          var reactComponent = #{react_component}(#{data_variable});
-          React.render(reactComponent, domNode);
+          React.render(#{render_js_react_element(react_component, data_variable)}, domNode);
```

Comment:
The JS code will be more readable if you breakout the reactComponent into a var.